draft-yasskin-httpbis-origin-signed-exchanges-impl.md

```diff
@@ -637,71 +889,147 @@ described here.
-':url' key's value.
 
-The second element of the array is interpreted as the exchange's response
-headers with lowercase names, with the 3-digit response status code in the
-':status' key's value.
 
-If any header field name includes uppercase characters, parsing MUST fail.
+## HTTP/2 extension for cross-origin Server Push # {#cross-origin-push}
+
+Cross origin push is not implemented.
+
+## application/signed-exchange format # {#application-signed-exchange}
 
-Pass the `Signature` response header and the exchange with that header removed
-to the algorithm in {{cross-origin-trust}}. Fail if this returns "invalid".
+To allow signed exchanges to be the targets of `<link rel=prefetch>` tags, we
+define the  `application/signed-exchange` content type that represents a signed
+HTTP exchange, including request metadata and header fields, response metadata
+and header fields, and a response payload.
+
+This content type consists of the concatenation of the following items:
+
+1. The ASCII characters "sxg1" followed by a 0 byte, to serve as a file
+   signature. This is redundant with the MIME type, and recipients that receive
+   both MUST check that they match and stop parsing if they don't.
+
+   Note: RFC EDITOR PLEASE DELETE THIS NOTE; The implementation of the final RFC
+   MUST use this file signature, but implementations of drafts MUST NOT use it
+   and MUST use another implementation-specific string beginning with "sxg1-" and
+   ending with a 0 byte instead.
+1. 3 bytes storing a big-endian integer `sigLength`. If this is larger
+   than TBD, parsing MUST fail.
+1. 3 bytes storing a big-endian integer `headerLength`. If this is larger than
+   TBD, parsing MUST fail.
+1. `sigLength` bytes holding the `Signature` header field's value
+   ({{signature-header}}).
+1. `headerLength` bytes holding the signed headers, the canonical serialization
+   ({{canonical-cbor}}) of the CBOR representation of the request and response
+   headers of the exchange represented by the `application/signed-exchange`
+   resource ({{cbor-representation}}), excluding the `Signature` header field.
+
+   Note that this is exactly the bytes used when checking signature validity in
+   {{signature-validity}}.
+1. The payload body (Section 3.3 of {{!RFC7230}}) of the exchange represented by
+   the `application/signed-exchange` resource.
+
+   Note that the use of the payload body here means that a `Transfer-Encoding`
+   header field inside the `application/signed-exchange` header block has no
+   effect. A `Transfer-Encoding` header field on the outer HTTP response that
+   transfers this resource still has its normal effect.
+
+### Cross-origin trust in application/signed-exchange {#co-trust-app-signed-exchange}
+
+To determine whether to trust a cross-origin exchange stored in an
+`application/signed-exchange` resource, pass the `Signature` header field from
+the non-signed header section and an exchange consisting of the headers from the
```

Comment:
Should this be updated to remove the reference to the "non-signed header section", or perhaps the above "`sigLength` bytes" item clarified to be named that?